src/teehr/loading/locations.py

```diff
@@ -151,40 +147,37 @@ def convert_locations(
 
 
 def validate_and_insert_single_locations(
-    conn: duckdb.DuckDBPyConnection,
     input_filepath: Union[str, Path],
     output_filepath: Union[str, Path],
 ):
     """Validate locations and copy to database."""
     logger.info(f"Validating and inserting locations from {input_filepath}")
 
-    insert_locations(conn, input_filepath)
+    gdf = gpd.read_parquet(input_filepath)
+
+    schema = pa.DataFrameSchema({
+        "id": pa.Column(str, coerce=True),
+        "name": pa.Column(str, coerce=True),
+        "geometry": pa.Column("geometry")
+    })
+
+    validated_gdf = schema.validate(gdf)
 
```

Comment:
> Do we need to check if errors were returned here, similar to the check in other the methods (location attributes, timeseries, crosswalk)?

I don't think so.  My understanding anyway is that the pandas/geopandas raise on failure which is different than how the pyspark dataframes work (I think because if being lazy maybe).  In the examples the catch the errors to prevent raise of error, but we actually went the other way and check for errors and raise where we use pyspark.  I might be wrong, but i think the way it is is consistent.

Reply:
Oh right gotcha, that makes sense